styleguide/writing-style.md

```diff
@@ -0,0 +1,183 @@
+
+### UI elements
+
+UI elements should be formatted in italics. For example:
+
+To get your agent token, navigate to _Agents_, then select _Reveal Agent Token_.
+
+> [!NOTE]
+> Match the capitalization used in the Buildkite interface, even if title (or any other) case has been used for product features.
+
+Refer to [UI elements in the Markdown syntax guide](markdown-syntax-style.md#ui-elements) for details on how to write and present UI elements in the Buildkite docs.
+
+### Lists (bullet lists and numbered steps)
+
+Capitalize the first word; no full stops at the end if it's not a full sentence. If it's a full sentence, give it a full stop.
+
+See also what [Google](https://developers.google.com/style/lists#capitalization-and-end-punctuation) and [Microsoft](https://learn.microsoft.com/en-us/style-guide/scannable-content/lists#punctuation) say about lists.
+
+### Writing numbers
+
+Write out numbers up to 10, then use digits - '58 bugs in this script and just two hours to fix them all!'
+Long numbers use commas to separate thousands - '100,000,000.00'. When in doubt, see what [Google](https://developers.google.com/style/numbers) and [Microsoft](https://docs.microsoft.com/en-us/style-guide/numbers) say about numbers.
+
+Do not use numbers in page headings. Only use numbers less than 10 in section headings.
+No restrictions on using numbers in the body of the text.
+
+### Time and date
+
+Use 24hr time with hours and minutes, but not seconds. Include timezone. For example, 17:00 AEST
+
+More about this in the [Google developer documentation style guide](https://developers.google.com/style/dates-times).
+
+### Spacing after full stops
+
+Refer to [Spacing after full stops in the Markdown syntax guide](markdown-syntax-style.md#spacing-after-full-stops) for more details about this.
+
+### Platform differences
+
+This table summarizes writing style differences across different platforms, distinguishing them from the Buildkite 'Docs'.
+
+|                      | Docs                                                      | Twitter and Blog                                                        | Changelog                                                               |
+|----------------------|-----------------------------------------------------------|-------------------------------------------------------------------------|-------------------------------------------------------------------------|
+| We (as in Buildkite) | No                                                        | Yes                                                                     | Avoid if possible; use 'I' if you need to                               |
+| Exclamation marks    | No                                                        | Yes, although use with restraint, and not more than one in a row        | Yes, although use with restraint, and not more than one in a row        |
```

Comment:
I agree with the sentiment, but wonder if this is too restrictive. I avoid "we" in almost every case, but do use it with "We recommend..." Similarly with exclamation marks, using one in a tutorial to celebrate success feels appropriate. What do you think about those two cases?

Reply:
Happy to make some exceptions here and there (and I'm happy with this one too), although ideally we should spell these out in the style guide.

Reply:
Michael and I agreed - simplify table to remove Twitter and changelog entries and provide generic guidance of usage of 'we' and exclamation marks (that is, avoid) unless tone of writing warrants it, such as a tutorial.

Reply:
Transformed into a new section on 'formality'.